aiida/cmdline/commands/cmd_group.py

```diff
@@ -57,6 +57,25 @@ def group_remove_nodes(group, nodes, clear, force):
         group.remove_nodes(nodes)
 
 
+@verdi_group.command('move-nodes')
+@options.SOURCE_GROUP(required=True)
+@options.TARGET_GROUP(required=True)
+@arguments.NODES()
+@options.FORCE()
+@with_dbenv()
+def group_move_nodes(source_group, target_group, nodes, force):
+    """Move nodes from one group to another."""
+    if not force:
+        click.confirm(
+            f'Do you really want to move {len(nodes)} nodes from Group<{source_group.label}> to '
+            f'Group<{target_group.label}>?',
+            abort=True
+        )
+
+    source_group.remove_nodes(nodes)
+    target_group.add_nodes(nodes)
```

Comment:
Maybe I'm being super paranoid, but if `add_nodes` fails after `remove_nodes` then you just lost track of your nodes, right? Would perhaps be slightly better to do it the other way around and/or `Try` to catch the errors and issue a warning saying the groups might have been messed up (or more specifically if you wrap each on a different `Try`)?

Reply:
Hmm, I don't immediately see a case where `add_nodes` would fail if `remove_nodes` already succeeded, but you're probably right to be paranoid. 😄 

Doing it the other way round seems more dangerous, e.g. if the user wants to move nodes that are not in the `source_group`, the command would add these nodes to the `target_group` and then fail to remove them from the `source_group`. In fact, I just noticed that `remove_nodes(nodes)` shows no error or warning when one of the `nodes` is not in the list of nodes, so we should probably fix that first. Let me open an issue for this.

I could use `try...except` to add a warning, but I wouldn't know which error to except, and if an error is raised, wouldn't the user be aware that the groups might have been messed up?

Reply:
> Doing it the other way round seems more dangerous, e.g. if the user wants to move nodes that are not in the source_group, the command would add these nodes to the target_group and then fail to remove them from the source_group. In fact, I just noticed that remove_nodes(nodes) shows no error or warning when one of the nodes is not in the list of nodes, so we should probably fix that first. Let me open an issue for this.

This is a very good point. I'm generally a big proponent of "programs should raise and not just ignore you when you ask them to do something they can't do" (unless you `-force`), but I think not all people share this view.

Maybe we need to do some checks before executing this commands? (i.e. that all nodes are in the source_group and none are in the target group? do we just need to check if they are in the source group? I'm not sure which behaviour we expect from this "mv" now =S)

> I could use try...except to add a warning, but I wouldn't know which error to except, and if an error is raised, wouldn't the user be aware that the groups might have been messed up?

Do we have a way to "append" to the error message? That way you can append something like "WARNING: Nodes have already been added to the target group" that will show at the end of the raise at a failed posterior "delete".

Reply:
Perhaps we should ask @giovannipizzi what he would expect of this command? Now I noticed his comment in #4426 mentions something related to this.

Reply:
Another question is at which level we want to do this check. Just at the level of the CLI command? Or maybe at the `GroupBackend` level? Should `group.remove_nodes(nodes)` not at least warn the user in case one of the nodes is not in the group?

Reply:
Yes, I know what you mean. Again, I would tipically be in favour of warning or even raising when you are asking to delete stuff that is not there. The only problem I see with this change is that it has the potential of being highly disruptive (meaning: backwards incompatible). I would think that for now we just have to accept that `remove_nodes` works this way and in any case start considering if this is a change we would want for 2.0. For now I would just reduce this discussion to how this "move" command behaves.

Reply:
Mmm, I think that in many cases it's good to 'stop/raise' if there is something really wrong (but maybe in a 'atomic' way, without having performed only half of the job?) but there are cases where this is not ideal (e.g. when a code "crashes" because some convergence loop and you don't even get partial information).
Here, for the internal API, I am not sure if we want to change the behaviour as this is indeed backward incompatible.
But for the CLI at least we should echo if something is unexpected.
Also, different tools behave differently - e.g. `mv` warns but continues, but `git mv` is much more strict.

Suggestion: propose the behaviour at the next AiiDA meeting and see if there is consensus.